Makefile

```diff
@@ -0,0 +1,95 @@
+client_start:
+	docker-compose -f build/deployments/docker-compose.yaml up -d client
+
+.PHONY: client_connect
+## Connect to the running client debugging daemon
+client_connect:
+	docker exec -it client /bin/bash -c "go run cmd/client/*.go"
+
+.PHONY: compose_and_watch
+## Run a localnet composed of 4 consensus validators w/ hot reload & debugging
+compose_and_watch:
+	docker-compose -f build/deployments/docker-compose.yaml up --force-recreate node1.consensus node2.consensus node3.consensus node4.consensus
+
+.PHONY: compose_and_watch
+## Kill all containers started by the docker-compose file
+docker_kill_all:
+	docker-compose -f build/deployments/docker-compose.yaml down
+
+.PHONY: docker_wipe
+## [WARNING] Remove all the docker containers, images and volumes.
+docker_wipe: prompt_user
+	docker ps -a -q | xargs -r -I {} docker stop {}
+	docker ps -a -q | xargs -r -I {} docker rm {}
+	docker images -q | xargs -r -I {} docker rmi {}
+	docker volume ls -q | xargs -r -I {} docker volume rm {}
+
+.PHONY: mockgen
+## Use `mockgen` to generate mocks used for testing purposes of all the modules.
+mockgen:
+	$(eval modules_dir = "shared/modules")
+	mockgen --source=${modules_dir}/persistence_module.go -destination=${modules_dir}/mocks/persistence_module_mock.go -aux_files=pocket/${modules_dir}=${modules_dir}/module.go
+	mockgen --source=${modules_dir}/p2p_module.go -destination=${modules_dir}/mocks/p2p_module_mock.go -aux_files=pocket/${modules_dir}=${modules_dir}/module.go
+	mockgen --source=${modules_dir}/utility_module.go -destination=${modules_dir}/mocks/utility_module_mock.go -aux_files=pocket/${modules_dir}=${modules_dir}/module.go
+	mockgen --source=${modules_dir}/consensus_module.go -destination=${modules_dir}/mocks/consensus_module_mock.go -aux_files=pocket/${modules_dir}=${modules_dir}/module.go
+	echo "Mocks generated in ${modules_dir}/mocks"
+
+.PHONY: test_all
+## Run all go unit tests
+test_all: # generate_mocks
+	go test ./...
+
+# TODO(olshansky): Replace `prefix` with `./` when moving over the protobufs.
+.PHONY: protogen_local
+## V1 Integration - Use `protoc` to generate consensus .go files from .proto files.
+protogen_local:
+	$(eval prefix = "./prototype")
+	$(eval proto_dir = "${prefix}/shared/types/proto/")
+	protoc -I=${proto_dir} --go_out=./ ${proto_dir}/*.proto
+	protoc -I=${proto_dir} -I=${prefix}/persistence/pre_persistence/proto --go_out=./ ${prefix}/persistence/pre_persistence/proto/*.proto
+	protoc -I=${proto_dir} -I=${prefix}/p2p/pre_p2p/types/proto --go_out=./ ${prefix}/p2p/pre_p2p/types/proto/*.proto
+	protoc -I=${proto_dir} -I=${prefix}/utility/proto --go_out=./ ${prefix}/utility/proto/*.proto
+	protoc -I=${proto_dir} -I=${prefix}/consensus/types/proto --go_out=./ ${prefix}/consensus/types/proto/*.proto
+	echo "View generated proto files by running: $ find . -name \"*.pb.go\" | grep -v \"./protoype\""
+
+.PHONY: protogen_m1
+## TODO(derrandz): Test, validate & update.
+protogen_m1:
```

Comment:
I renamed it to `protogen_docker` but note the TODO I left for Hamza since it wasn't working for me and is also outside the scope of this PR.

I'm sticking with `protogen_local` to unblock the app-specific bus implementation.